packages/web/webpack/common.ts

```diff
@@ -0,0 +1,28 @@
+import { resolve } from 'path';
+import { Configuration, DefinePlugin } from 'webpack';
+import MiniCssExtractPlugin from 'mini-css-extract-plugin';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const SpritePlugin = require('svg-sprite-loader/plugin');
+
+export const srcDir = 'src';
+
+export const configCommon: (mode: 'development' | 'production') => Configuration = (mode) => ({
+  mode,
+  resolve: {
+    extensions: ['.ts', '.tsx', '.js'],
+    modules: [resolve(__dirname, '..', srcDir), 'node_modules'],
+  },
+  devtool: mode === 'development' ? 'eval-source-map' : false,
+});
+
+export const pluginsCommon = (mode: string, isServer: boolean) => [
+  new DefinePlugin({
+    __isServer__: isServer,
+    ...(isServer ? {} : {'process.env': 'window.env'}),
```

Comment:
We can always refer to `process.env` global, while assigning env vars as `window.process = { env: {...} }` client-side. IMO, this is same amount of magic, but less clutter overall.